components/external-services-mock/internal/selfreg/handler.go

```diff
@@ -70,15 +75,36 @@ func (h *Handler) HandleSelfRegPrep(w http.ResponseWriter, r *http.Request) {
 		httphelpers.WriteError(w, err, http.StatusBadRequest)
 		return
 	}
+
+	SelfRegistrations[name] = tenant
 }
 
 func (h *Handler) HandleSelfRegCleanup(w http.ResponseWriter, r *http.Request) {
 	params := mux.Vars(r)
-	if name, ok := params[NamePath]; !ok || name == "" {
+	name, ok := params[NamePath]
+	if !ok || name == "" {
 		err := errors.New(fmt.Sprintf("%s is missing from path", NamePath))
 		httphelpers.WriteError(w, err, http.StatusBadRequest)
 		return
 	}
+
+	providerSubaccount, cloningExists := SelfRegistrations[name]
+
+	subscriberExists := false
+	for _, provider := range subscription.Subscriptions {
+		if provider == providerSubaccount {
+			subscriberExists = true
+			break
+		}
+	}
+
+	if cloningExists && subscriberExists {
+		// We swallow their error msg and print only the status code
+		err := errors.New("")
+		httphelpers.WriteError(w, err, http.StatusConflict)
+		return
```

Comment:
If you adapt the comment above you could put this in the for cycle and remove this if